.github/actions/latest-stable-tag/get-latest-tag.js

```diff
@@ -0,0 +1,41 @@
+export function getLatestTag(tags) {
+  const sortedTags = tags
+    .filter(tag => !isPreRelease(tag))
+    .sort(compareTagsDescending);
+  const latestTag = sortedTags[0];
+  return latestTag;
+}
+
+function isPreRelease(tag) {
+    return tag.includes('-');
+}
+
+function compareTagsDescending(tag1, tag2) {
+    return compareTags(tag2, tag1);
+}
+
+function compareTags(tag1, tag2) {
+    // Note: assume major.minor.patch format
+    // Pre-release tags should have already been excluded
+    const parts1 = tag1.split('.');
+    const parts2 = tag2.split('.');
+
+    for (let i = 0; i < 3; i++) {
+        let v1 = parseInt(parts1[i]);
+        let v2 = parseInt(parts2[i]);
+        if (v1 && v2) {
+            if (v1 < v2) return -1;
+            if (v1 > v2) return 1;
+        } else if (v1 && !v2) {
+            // tag1 is longer than tag2, consider tag1 greater
```

Comment:
Or `tag2` has 0 as this component and tag1 doesn't. tag1 is still greater, but not for the reason given in the comment.

(I am saying this because I believe that 0 is truthy in javascript…)

Reply:
While I'm here, you have code that considers different-length tags, but no tests for them.

Reply:
> I believe that 0 is truthy in javascript

No, it's falsy. And it makes me question the way I wrote this code. I think I meant to check whether `v1` and `v2` were defined... I'll double check

> While I'm here, you have code that considers different-length tags, but no tests for them.

Good catch!

Reply:
OK, so `parseInt` actually _never_ returns `undefined`. It either returns a valid result, or `NaN`, which is falsy. But `0` is also falsy...

I added this test case, which fails:

```javascript
const TAGS_WITH_ZERO_MAJOR = [ '0.0.1', '0.1.2', '0.1.0' ];
```

It should return `0.1.2`, but actually returns `0.0.1`, because it considers them all equal...

Reply:
Oh, I 100% meant to write "falsey". 😊